test-tests/shared/src/test/scala/zio/test/environment/ClockSpec.scala

```diff
@@ -0,0 +1,156 @@
+            _         <- latch1.await
+            _         <- testClock.adjust(2.hours)
+            _         <- latch2.await
+            result    <- ref.get
+          } yield assert(result, equalTo("Hello, World!"))
+        } @@ nonFlaky(100),
+        testM("sleep correctly handles new set time") {
+          for {
+            latch <- Promise.make[Nothing, Unit]
+            _     <- sleep(10.hours).flatMap(_ => latch.succeed(())).fork
+            _     <- setTime(11.hours)
+            _     <- latch.await
+          } yield assert((), anything)
+        } @@ nonFlaky(100),
+        testM("sleep does sleep instantly when sleep duration less than set time") {
+          for {
+            testClock <- TestClock.makeTest(DefaultData)
+            latch     <- Promise.make[Nothing, Unit]
+            _         <- (testClock.setTime(11.hours) *> latch.succeed(())).fork
+            _         <- latch.await *> testClock.sleep(10.hours)
+          } yield assert((), anything)
+        } @@ nonFlaky(100),
+        testM("adjust correctly advances nanotime") {
+          for {
+            time1 <- nanoTime
+            _     <- adjust(1.millis)
+            time2 <- nanoTime
+          } yield assert(fromNanos(time2 - time1), equalTo(1.millis))
+        },
+        testM("adjust correctly advances currentTime") {
+          for {
+            time1 <- currentTime(TimeUnit.NANOSECONDS)
+            _     <- adjust(1.millis)
+            time2 <- currentTime(TimeUnit.NANOSECONDS)
+          } yield assert(time2 - time1, equalTo(1.millis.toNanos))
+        },
+        testM("adjust correctly advances currentDateTime") {
+          for {
+            time1 <- currentDateTime
+            _     <- adjust(1.millis)
+            time2 <- currentDateTime
+          } yield assert((time2.toInstant.toEpochMilli - time1.toInstant.toEpochMilli), equalTo(1L))
+        },
+        testM("adjust does not produce sleeps") {
+          adjust(1.millis)
+          assertM(sleeps, isEmpty)
+        },
+        testM("setTime correctly sets nanotime") {
+          for {
+            _    <- setTime(1.millis)
+            time <- clock.nanoTime
+          } yield assert(time, equalTo(1.millis.toNanos))
+        },
+        testM("setTime correctly sets currentTime") {
+          for {
+            _    <- setTime(1.millis)
+            time <- currentTime(TimeUnit.NANOSECONDS)
+          } yield assert(time, equalTo(1.millis.toNanos))
+        },
+        testM("setTime correctly sets currentDateTime") {
+          for {
+            _    <- TestClock.setTime(1.millis)
+            time <- currentDateTime
+          } yield assert(time.toInstant.toEpochMilli, equalTo(1.millis.toMillis))
+        },
+        testM("setTime does not produce sleeps") {
+          for {
+            _      <- setTime(1.millis)
+            sleeps <- sleeps
+          } yield assert(sleeps, isEmpty)
+        },
+        testM("setTimeZone correctly sets timeZone") {
```

Comment:
I don't think this test is working correctly right now because setting the time is an effect so it needs to be chained in a `flatMap` or a for comprehension to the assertion. The test is just passing right now because the default time zone is "UTC".

Reply:
i still have my imperative hat on :(